design/ceph/object/ceph-sse-s3.md

```diff
@@ -0,0 +1,42 @@
+---
+# AWS server side encryption SSE-S3 support for RGW
+target-version: release-1.10
+---
+
+# Feature Name
+AWS server side encryption SSE-S3 support for RGW
+
+## Summary
+The S3 protocol over three different types of [server side encryption](https://docs.aws.amazon.com/AmazonS3/latest/userguide/serv-side-encryption.html): SSE-C, SSE-KMS and SSE-S3. For the last two RGW server need to configure with external services such as [vault](https://www.vaultproject.io/). Currently Rook configure RGW with `SSE-KMS` options, so that S3 request with `sse:kms` header will be handled properly. Recently the support for handling the `sse:s3` added to RGW with another set of options, so Rook needs to provide option to configure RGW similar to `SSE-KMS`.
```

Comment:
> Currently Rook configure RGW with `SSE-KMS` options, so that S3 request with `sse:kms` header will be handled properly.

I understand from this that Ceph RGW currently supports sse:s3, and that those details are configured by Rook's `spec.security.kms`.

> Recently the support for handling the `sse:s3` added to RGW with another set of options, so Rook needs to provide option to configure RGW similar to `SSE-KMS`.

Will we need to do conditional handling in Rook? For example, will we need to tell users to use only the `kms` option for Ceph versions below v17 (or some other version)? If so, please capture that specific detail. At any rate, we need to note the Ceph version that adds support for `sse:s3` so we can be sure to document that the option doesn't apply for some Ceph versions.